src/auth/Auth.cpp

```diff
@@ -0,0 +1,124 @@
+#include "Auth.hpp"
+#include "Pam.hpp"
+#include "Fingerprint.hpp"
+#include "../config/ConfigManager.hpp"
+#include "../core/hyprlock.hpp"
+#include "src/helpers/Log.hpp"
+
+#include <hyprlang.hpp>
+#include <memory>
+
+CAuth::CAuth() {
+    static auto* const PENABLEPAM = (Hyprlang::INT* const*)g_pConfigManager->getValuePtr("auth:pam:enabled");
+    if (**PENABLEPAM)
+        m_vImpls.push_back(std::make_shared<CPam>());
+    static auto* const PENABLEFINGERPRINT = (Hyprlang::INT* const*)g_pConfigManager->getValuePtr("auth:fingerprint:enabled");
+    if (**PENABLEFINGERPRINT)
+        m_vImpls.push_back(std::make_shared<CFingerprint>());
+
+    RASSERT(!m_vImpls.empty(), "At least one authentication method must be enabled!");
+}
+
+void CAuth::start() {
+    for (const auto& i : m_vImpls) {
+        i->init();
+    }
+}
+
+void CAuth::submitInput(const std::string& input) {
+    for (const auto& i : m_vImpls) {
+        i->handleInput(input);
+    }
+}
+
+bool CAuth::isAuthenticated() {
+    for (const auto& i : m_vImpls) {
+        if (i->isAuthenticated())
+            return true;
+    }
+
+    return false;
+}
+
+bool CAuth::checkWaiting() {
+    for (const auto& i : m_vImpls) {
+        if (i->checkWaiting())
+            return true;
+    }
+
+    return false;
+}
+
+std::string CAuth::getInlineFeedback() {
+    std::optional<std::string> firstFeedback = std::nullopt;
+    for (const auto& i : m_vImpls) {
+        const auto FEEDBACK = (m_bDisplayFailText) ? i->getLastFailText() : i->getLastPrompt();
+        if (!FEEDBACK.has_value())
+            continue;
+
+        if (!firstFeedback.has_value())
+            firstFeedback = FEEDBACK;
+
+        if (i->getImplType() == m_eLastActiveImpl)
+            return FEEDBACK.value();
+    }
+
+    return firstFeedback.value_or("Ups, no authentication feedack");
+}
+
+std::optional<std::string> CAuth::getFailText(eAuthImplementations implType) {
+    for (const auto& i : m_vImpls) {
+        if (i->getImplType() == implType) {
+            return i->getLastFailText();
+        }
+    }
+    return std::nullopt;
+}
+
+std::optional<std::string> CAuth::getPrompt(eAuthImplementations implType) {
+    for (const auto& i : m_vImpls) {
+        if (i->getImplType() == implType) {
```

Comment:
oopsie, missed these two (this one and above)

Reply:
a shame clang-format `RemoveBracesLLVM` removes `for` braces as well.
do you have any linter check for that?

Reply:
nope